desktop-app/app/components/DeviceManager/index.js

```diff
@@ -45,9 +45,26 @@ export default function DeviceManager(props) {
       acc[val.id] = val;
       return acc;
     }, {});
-    const inactiveDevices = props.browser.allDevices.filter(
-      device => !activeDevicesById[device.id]
-    );
+
+    const currentInactiveDevicesById = devices.inactive.reduce((acc, val) => {
+      acc[val.id] = val;
+      return acc;
+    }, {});
+
+    const devicesById = props.browser.allDevices.reduce((acc, val) => {
+      acc[val.id] = val;
+      return acc;
+    }, {});
+
+    const inactiveDevices = [
+      ...props.browser.allDevices.filter(
```

Comment:
During testing I saw this expression only returned something during the first render. If that is the case all the work done to generate this value in subsequent renders is unnecessary. I would recommend to move that work to a `useMemo` that only runs once.

Reply:
@jzabala Are you considereing the scenario where the user adds or removes devices ?, In those scenarios the `browser.allDevices` changes right?

Reply:
Yeah, but it is added to the active devices making the expression false at line 62.

Reply:
You can add new devices to 'inactive' list directly too

Reply:
![image](https://user-images.githubusercontent.com/13673443/91323861-ab7cab00-e7c1-11ea-8575-519e157b1d08.png)

Reply:
Is it an expected behavior that Custom devices in the inactive list always get to the top when I close and open the app?